src/index.js

```diff
@@ -75,7 +75,7 @@ HTTPSnippet.prototype.prepare = function (request) {
   if (request.headers && request.headers.length) {
     // loweCase header keys
     request.headersObj = request.headers.reduce(function (headers, header) {
-      headers[header.name.toLowerCase()] = header.value
+      headers[header.name] = header.value
```

Comment:
There is a comment in the linked issue mentioning that HTTP/2 _requires_ headers to be lowercase, while they are case-sensitive in HTTP/1.1 - https://github.com/Kong/httpsnippet/issues/74#issuecomment-561412157.

That's a great point, and since we have access to the version here (`request.httpVersion`) I am inclined suggest only converting to lowercase with HTTP/2, and use `header.name` as-is with other cases.

Reply:
Makes sense to me. @jgiovaresco can you add `httpVersion` detection and only lowercase on http/2.x?

re:
> Malformed Requests and Responses
A malformed request or response is one that is an otherwise valid
sequence of HTTP/2 frames but is invalid due to the presence of
extraneous frames, prohibited header fields, the absence of mandatory
header fields, or the inclusion of uppercase header field names.